gammapy/maps/hpx/ndmap.py

```diff
@@ -405,6 +405,41 @@ def cutout(self, position, width, *args, **kwargs):
         data = self.data[..., idx]
         return self.__class__(geom=geom, data=data, unit=self.unit, meta=self.meta)
 
+    def _cutout_view(self, position, width, odd_npix=False):
+        """
+        Create a cutout around a given position without copy of the data.
+
+        Parameters
+        ----------
+        position : `~astropy.coordinates.SkyCoord`
+            Center position of the cutout region.
+        width : tuple of `~astropy.coordinates.Angle`
+            Angular sizes of the region in (lon, lat) in that specific order.
+            If only one value is passed, a square region is extracted.
+        odd_npix : bool, optional
+            Force width to odd number of pixels.
+            Default is False.
+
+        Returns
+        -------
+        cutout : `~gammapy.maps.HpxNDMap`
+            Cutout map.
+        """
+        geom_cutout = self.geom.cutout(
+            position=position, width=width, mode="trim", odd_npix=odd_npix
+        )
+
+        if self.geom.is_allsky:
+            idx = geom_cutout._ipix
+        else:
+            idx = self.geom.to_image().global_to_local((geom_cutout._ipix,))[0]
+
+        data = self.quantity[..., idx]
```

Comment:
There is also a fundamental problem, the fancy indexing return a copy and not a view. As for HpxMap the indexes are not always contiguous we cannot rewrite that in term of slices that give a view as for the  WcsMap.
So I think we cannot define `_cutout_view` in that case, and the fix would rather be a try/except statement in the MapEvaluator that fall back to cutout if _cutout_view is not defined.